rspack/crates/binding/Cargo.toml

```diff
@@ -0,0 +1,52 @@
+[package]
+name              = "next-rspack-binding"
+publish           = false
+edition           = "2024"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+rspack_binding_builder        = { version = "=0.4.10" }
+rspack_binding_builder_macros = { version = "=0.4.10" }
+
+rspack_core             = { version = "=0.4.10" }
+rspack_error            = { version = "=0.4.10" }
+rspack_hook             = { version = "=0.4.10" }
+rspack_plugin_externals = { version = "=0.4.10" }
+rspack_sources          = { version = "=0.4.8" }
+
+regex        = { version = "1.11.1" }
+rspack_regex = { version = "=0.4.10" }
+rustc-hash   = { version = "2.1.1" }
+
+napi        = { version = "=3.1.2" }
+napi-derive = { version = "=3.1.1" }
+
+next-taskless = { path = "../../../crates/next-taskless" }
+
+# Enable SWC plugin feature for targets that support it
+# Skip: wasm32-wasip1-threads, i686-pc-windows-msvc, aarch64-pc-windows-msvc, armv7-linux-androideabi, armv7-unknown-linux-gnueabihf
+[target.'cfg(not(any(target_arch = "wasm32", target_arch = "arm", all(target_os = "windows", target_arch = "x86"), all(target_os = "windows", target_arch = "aarch64"))))'.dependencies]
+rspack_binding_builder = { version = "=0.4.10", features = ["plugin"] }
+
+[build-dependencies]
+rspack_binding_build = { version = "=0.4.10" }
+
+# Copied from https://github.com/web-infra-dev/rspack/blob/main/Cargo.toml
+
+[profile.dev]
+codegen-units = 16
+debug         = 2
+incremental   = true
+panic         = "abort"
+
+[profile.release]
+codegen-units = 1
+debug         = false
+# Performs “fat” LTO which attempts to perform optimizations across all crates within the dependency graph.
+lto       = "fat"
```

Comment:
I'm curious to know how much benefit you get from this. We considered it for Turbopack, but decided against it for now because the increase to build time was massive (we do release builds on every PR as part of the CI workflows) and the performance improvement was really minimal (a couple percent?).

Reply:
I've found the related PR here: https://github.com/web-infra-dev/rspack/pull/7088. 

Better performance (according to benchmark): 1% ~ 3% faster
Smaller binary size (rspack.darwin-arm64.node): 55.4 MB -> 51.7 MB

Longer build time, only affect the production release: 4-5min -> 8-10min

Reply:
Okay, we can do this for now, but we might remove it later.